source/credentials_provider_sts_web_identity.c

```diff
@@ -0,0 +1,1120 @@
+            }
+
+            if ((!role_arn || !role_arn->len)) {
+                if (role_arn) {
+                    aws_string_destroy(role_arn);
+                }
+                struct aws_profile_property *role_arn_property =
+                    aws_profile_get_property(profile, s_sts_web_identity_role_arn_config);
+                if (role_arn_property) {
+                    role_arn = aws_string_new_from_string(allocator, role_arn_property->value);
+                }
+            }
+            if ((!role_session_name || !role_session_name->len)) {
+                if (role_session_name) {
+                    aws_string_destroy(role_session_name);
+                }
+                struct aws_profile_property *role_session_name_property =
+                    aws_profile_get_property(profile, s_sts_web_identity_role_session_name_config);
+                if (role_session_name_property) {
+                    role_session_name = aws_string_new_from_string(allocator, role_session_name_property->value);
+                }
+            }
+            if ((!token_file_path || !token_file_path->len)) {
+                if (token_file_path) {
+                    aws_string_destroy(token_file_path);
+                }
+                struct aws_profile_property *token_file_path_property =
+                    aws_profile_get_property(profile, s_sts_web_identity_token_file_path_config);
+                if (token_file_path_property) {
+                    token_file_path = aws_string_new_from_string(allocator, token_file_path_property->value);
+                }
+            }
+        }
+    }
+
+    struct aws_byte_cursor region_cursor;
+    if (region && region->len) {
+        region_cursor = aws_byte_cursor_from_array(region->bytes, region->len);
+    } else {
+        region_cursor = s_default_region_cursor;
+    }
+
+    /* determine endpoint */
+    if (aws_byte_buf_init_copy_from_cursor(&parameter->endpoint, allocator, s_sts_cursor)) {
+        goto on_finish;
+    }
+    if (aws_byte_buf_append_dynamic(&parameter->endpoint, &region_cursor)) {
+        goto on_finish;
+    }
+    if (aws_byte_buf_append_dynamic(&parameter->endpoint, &s_amazonaws_cursor)) {
+        goto on_finish;
+    }
+
+    if (aws_string_eq_c_str_ignore_case(region, "cn-north-1") ||
+        aws_string_eq_c_str_ignore_case(region, "cn-northwest-1")) {
+        if (aws_byte_buf_append_dynamic(&parameter->endpoint, &s_cn_cursor)) {
+            goto on_finish;
+        }
+    }
+
+    /* determine role_arn */
+    if (!role_arn || !role_arn->len) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Failed to resolve role arn during sts web identity provider initialization.")
+        goto on_finish;
+    }
+
+    struct aws_byte_cursor role_arn_cursor = aws_byte_cursor_from_string(role_arn);
+    if (aws_byte_buf_init_copy_from_cursor(&parameter->role_arn, allocator, role_arn_cursor)) {
+        AWS_LOGF_ERROR(
```

Comment:
Don't think this logging is needed, it's essentially an OOM.  Similar for the other buf init failures.